include/GafferImage/ImageReader.h

```diff
@@ -110,10 +110,18 @@ class ImageReader : public ImageNode
 		Gaffer::StringPlug *colorSpacePlug();
 		const Gaffer::StringPlug *colorSpacePlug() const;
 
+		Gaffer::StringPlug *colorSpaceMetadataNamePlug();
+		const Gaffer::StringPlug *colorSpaceMetadataNamePlug() const;
+
 		virtual void affects( const Gaffer::Plug *input, AffectedPlugsContainer &outputs ) const;
 
 		static size_t supportedExtensions( std::vector<std::string> &extensions );
 
+		typedef boost::function<const std::string ( const IECore::CompoundData * )> DefaultColorSpaceFunction;	
+		static void registerDefaultColorSpace( DefaultColorSpaceFunction f );
```

Comment:
the dataType in this case comes from the ImageSpec itself, I'm using the `format` data member.